TSPL.docc/LanguageGuide/Concurrency.md

```diff
@@ -564,33 +637,192 @@ the explicit parent-child relationships between tasks
 let Swift handle some behaviors like propagating cancellation for you,
 and lets Swift detect some errors at compile time.
 
+[`TaskGroup`]: https://developer.apple.com/documentation/swift/taskgroup
+
+<!-- TR: What's a good example of this kind of error? -->
+
+Here's another version of the code to download photos
+that handles any number of photos:
+
 ```swift
 await withTaskGroup(of: Data.self) { taskGroup in
+    let photoNames = await listPhotos(inGallery: "Summer Vacation")
+    for name in photoNames {
+        taskGroup.addTask {
+            let photo = downloadPhoto(named: name)
+            show(photo)
+        }
+    }
+}
+```
+
+The code above creates a new task group,
+and then creates child tasks of that task group
+to download each photo is the gallery.
+Each task group downloads a single photograph
+and then adds it to the photographs being displayed.
+Because there isn't a specific priority passed
+when calling `addTask(priority:operation:)`,
+these child tasks tasks get the same priority as the task group.
+Swift schedules these tasks,
+running as many of them at the same time as conditions allow.
+
+> Note:
+> If the code to download a photo could throw an error,
+> you would call `withThrowingTaskGroup(of:returning:body:)` instead.
+
+In the code listing above,
+each photo is downloaded and immediately displayed,
+so the task group doesn't return any results.
```

Comment:
we have `withDiscardingTaskGroup` which is optimized for child tasks whose values we don't collect (which is what the current example is doing).

I'd suggest showing the `addTask + for await` pattern here, and we can later show the withDiscardingTaskGroup where we do the `show()` inside the `addTask` like the current example does

Reply:
Saved in a TODO for the future (428e3414).